policyengine_api/routes/household_routes.py

```diff
@@ -0,0 +1,137 @@
+from flask import Blueprint, Response, request
+from werkzeug.exceptions import NotFound, BadRequest
+import json
+
+from policyengine_api.services.household_service import HouseholdService
+from policyengine_api.utils.payload_validators import (
+    validate_household_payload,
+    validate_country,
+)
+
+
+household_bp = Blueprint("household", __name__)
+household_service = HouseholdService()
+
+
+@household_bp.route(
+    "/<country_id>/household/<int:household_id>", methods=["GET"]
+)
+@validate_country
+def get_household(country_id: str, household_id: int) -> Response:
+    """
+    Get a household's input data with a given ID.
+
+    Args:
+        country_id (str): The country ID.
+        household_id (int): The household ID.
+    """
+    print(f"Got request for household {household_id} in country {country_id}")
+
+    household: dict | None = household_service.get_household(
+        country_id, household_id
+    )
+    if household is None:
+        raise NotFound(f"Household #{household_id} not found.")
+    else:
```

Comment:
*nitt, style* - suggest shipping and optionally changing later to unblock me (please:P ), but clean code suggests 

```
if error:
  raise Exception("...")

return success 
```
instead of unnecessary else's or nesting.


If you don't have the book, [this is an ok summary](https://dev.to/shameel/-clean-coding-tip-early-return-principle-47kj)